tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtifactStepDefinitions.cs

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using BoDi;
+using Monai.Deploy.WorkflowManager.IntegrationTests.Support;
+using TechTalk.SpecFlow.Infrastructure;
+
+namespace Monai.Deploy.WorkflowManager.IntegrationTests.StepDefinitions
+{
+    [Binding]
+    public class WorkflowTaskArtifactStepDefinitions
+    {
+        private Assertions Assertions { get; set; }
+        private DataHelper DataHelper { get; set; }
+        private readonly ISpecFlowOutputHelper _outputHelper;
+
+        public WorkflowTaskArtifactStepDefinitions(ObjectContainer objectContainer, ScenarioContext scenarioContext, ISpecFlowOutputHelper outputHelper)
+        {
+            Assertions = new Assertions(objectContainer);
+            DataHelper = objectContainer.Resolve<DataHelper>();
+            _outputHelper = outputHelper;
+        }
+
+        [Then(@"Input artifacts are mapped")]
+        public void ThenInputArtifactsAreMapped()
+        {
+            string PayloadId;
+
+            if (DataHelper.WorkflowRequestMessage.PayloadId.ToString() != "00000000-0000-0000-0000-000000000000")
+            {
+                PayloadId = DataHelper.WorkflowRequestMessage.PayloadId.ToString();
+            }
+            else
+            {
+                PayloadId = DataHelper.WorkflowInstances[0].PayloadId;
+            }
+
+            _outputHelper.WriteLine($"Retrieving workflow instance using the payloadid={PayloadId}");
+            var workflowInstances = DataHelper.GetWorkflowInstances(1, PayloadId);
+            _outputHelper.WriteLine("Retrieved workflow instance");
+
+            if (workflowInstances != null)
+            {
+                foreach (var workflowInstance in workflowInstances)
+                {
+                    var workflowRevision = DataHelper.WorkflowRevisions.OrderByDescending(x => x.Revision).FirstOrDefault(x => x.WorkflowId.Equals(workflowInstance.WorkflowId));
+
+                    if (workflowRevision != null)
+                    {
+                        foreach (var task in workflowInstance.Tasks)
+                        {
+                            var seededTask = workflowInstance.Tasks.FirstOrDefault(x => x.ExecutionId.Equals(task.ExecutionId));
+
+                            if (seededTask == null)
+                            {
+                                var workflowTask = workflowRevision.Workflow.Tasks.FirstOrDefault(x => x.Id.Equals(task.TaskId));
+
+                                if (workflowTask != null)
+                                {
+                                    Assertions.AssertInputArtifacts(workflowTask, PayloadId, task);
+                                }
+                                else
+                                {
```

Comment:
This is indented quite a lot its worth learning about "inverted if" checking and "split loop And work"...

https://softwareengineering.stackexchange.com/questions/220461/keep-indentation-level-low

helps with cleaner code.

Reply:
Refactored